src/Intensities/Binning.jl

```diff
@@ -0,0 +1,438 @@
+
+"""
+    BinningParameters(binstart,binend,binwidth;covectors = I(4))
+    BinningParameters(binstart,binend;numbins,covectors = I(4))
+
+Describes a 4D parallelepided histogram in a format compatible with experimental Inelasitic Neutron Scattering data.
+See [`generate_mantid_script_from_binning_parameters`](@ref) to convert [`BinningParameters`](@ref) to a format understandable by the [Mantid software](https://www.mantidproject.org/), or [`load_nxs`](@ref) to load [`BinningParameters`](@ref) from a Mantid `.nxs` file.
+ 
+The coordinates of the histogram axes are specified by multiplication 
+of `(q,ω)` with each row of the `covectors` matrix.
+Since the `covectors` matrix is the identity matrix by default, the default coordinates
+are `(q,ω)` in Reciprocal Lattice Units (R.L.U.). 
+To use physical units instead, see [`rlu_to_absolute_units!`](@ref).
+
+The convention for the binning scheme is that:
+- The left edge of the first bin starts at `binstart`
+- The bin width is `binwidth`
+- The last bin contains `binend`
+- There are no "partial bins;" the last bin may contain values greater than `binend`. C.f. [`count_bins`](@ref).
+
+A `value` can be binned by computing its bin index:
+
+    coords = covectors * value
+    bin_ix = 1 .+ floor.(Int64,(coords .- binstart) ./ binwidth)
+"""
+mutable struct BinningParameters
```

Comment:
This is a very important point. I think what actually needs to happen is to make the `BinningParmaeters` in terms of physical (absolute) units instead of R.L.U., which are problem- and lattice-dependent (i.e. they aren't a real unit in any meaningful sense). As a bonus, no conversion is needed to go to Mantid.

Reply:
Sounds good. Doing all the binning code consistently in terms of absolute units makes sense to me.